### ISSUE TITLE
x86 support

### ISSUE THREAD

**Issue by Trass3r** [microsoft/llvm-mctoll#30]
Just curious, any plans for 32bit support?

**Comment by aaronsm** [comment]
For Intel we decided to go for x64 support. No plans for x32 right now but I agree it would be an interesting target to look at.

**Comment by Trass3r** [comment]
I see, is it even feasible or are there many hidden hard-coded assumptions?

**Comment by aaronsm** [comment]
It's definitely feasible. The disassembler is coming from LLVM which works for x32 so getting MCInst out is no problem. Once you have MCInst it's going to be different but the basic framework is there to follow.

**Comment by Necrolis** [comment]
Is there a road map of planned features or support? Might be helpful to those wanting to contribute to the project.

I personally would love to see Win PE/COFF and x86 support, and will likely tinker around with this project to see if I can get it compiling cleanly under Windows/VS and at least add COFF support. Even having a basic layout or flow of the internals would aid in grokking the internals a bit better.

**Comment by Trass3r** [comment]
Yes it compiles cleanly on Windows since it's just llvm. But it takes up to an hour.
You may use WSL to generate Linux test binaries.

**Comment by Necrolis** [comment]
@Trass3r might be helpful to get a wiki page on getting this working under Windows, though those compile times are a bit of a knock. Have you tried using incremental compilation to get them to more reasonable amounts?

**Comment by Trass3r** [comment]
It's all in the readme, though a bit convoluted. You can use the Visual Studio or Ninja generator.
I think it's like half an hour for a full build of only mctoll and an hour if you build clang. Incremental builds are faster of course.

**Comment by Necrolis** [comment]
@Trass3r I'm actually currently going through that (waiting for LLVM to clone), but one can immediately see that the readme would only work under WSL, where as I am talking under the native windows environment (though it doesn't much adaptation). Just nice to have a "Build Instructions (Windows)" section :) 

> They don't support PCH/modules or shared libraries on Windows.
I assume you by they you mean LLVM? I find that strange as I did have LLVM 4.x compiling both as a shared lib and static lib quite a while ago. I dunno how much has changed since then given that we are now at LLVM 9.x

**Comment by Trass3r** [comment]
Make sure to do a shallow clone `-b master --depth 1`.
Building llvm's components as shared libs never worked on Windows. If you mean libllvm with the c api, that's something else.

**Comment by Necrolis** [comment]
>Make sure to do a shallow clone -b master --depth 1.

Bit late for that, I am unfortunately not a git guru... 

I opened #37 for this seeing as this now more about windows support than x86 support.
